### PR TITLE
feat: implement rename table

### PR DESCRIPTION
**Pull request by e1ijah1**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

implement renaming table in mito table engine

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#723

### CODE REVIEW

File: src/datanode/src/sql/alter.rs [hunk 1]
```diff
@@ -76,13 +76,9 @@ impl SqlHandler {
...
                 names: vec![name.value.clone()],
             },
-            AlterTableOperation::RenameTable { .. } => {
-                // TODO update proto to support alter table name
-                return error::InvalidSqlSnafu {
-                    msg: "rename table not unsupported yet".to_string(),
-                }
-                .fail();
-            }
+            AlterTableOperation::RenameTable { new_table_name } => AlterKind::RenameTable {
```
Comment (evenyag) [inline src/datanode/src/sql/alter.rs:79]: You might need to fix this test: `test_alter_to_request_with_renaming_table`.

**Review by evenyag — COMMENTED** [review]
There are some conflicts that you might need to resolve

File: src/mito/src/table.rs [hunk 2]
```diff
@@ -419,6 +433,8 @@ fn create_alter_operation(
         AlterKind::DropColumns { names } => Ok(AlterOperation::DropColumns {
             names: names.to_vec(),
         }),
+        // No need to build alter operation when reaming tables.
+        AlterKind::RenameTable { .. } => unreachable!(),
```
[the remark this answers is not in the document]
Comment (e1ijah1) [reply src/mito/src/table.rs:437]: ok, I'll fix it

**Review by evenyag — APPROVED** [review]
LGTM. You might need to resolve the conflicts again 😢

**Comment by e1ijah1** [comment]
> LGTM. You might need to resolve the conflicts again 😢

It's all fixed now.

**Review by waynexia — APPROVED** [review]

**Pull request merged**